src/seedu/addressbook/Main.java

```diff
@@ -105,18 +113,29 @@ private void recordResult(CommandResult result) {
      * @param command user command
      * @return result of the command
      */
-    private CommandResult executeCommand(Command command)  {
+    private CommandResult executeCommand(Command command) {
+        command.setData(addressBook, lastShownList);
+        CommandResult result = command.execute();
         try {
-            command.setData(addressBook, lastShownList);
-            CommandResult result = command.execute();
+
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException se) {
+            ui.showToUser(storage.getPath() + message.MESSAGE_CANNOT_WRITE_FILE);
+            ui.showToUser(LINE_PREFIX + "Enter yes: ");
+            String confirmString = in.nextLine();
+            executeCommand(command);
```

Comment:
Logic is a bit confusing here. E.g.: Whats the role of confirmString? 
Can this part be made more self-explanatory?